src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -381,4 +376,31 @@ private CaseDetailsDTO buildDetailedCaseDTO(Case caze, boolean caseevents, boole
 
     return caseDetailsDTO;
   }
+
+  /**
+   * Deletes all case, caseevent and casegroup data for a particular collection exercise
+   *
+   * @param collectionExerciseId The Collection Exercise UUID to delete for
+   * @return An appropriate HTTP repsonse code
+   */
+  @DeleteMapping("{collectionExerciseId}")
```

Comment:
Its unfortunate that the existing API is a bit confusing, but I don't think this new endpoint is quite right.

There is a `GET` method is defined as  `/cases/{caseId}`. I don't think the equivalent `DELETE` method for that resource should accept a different parameter (i.e. the `collectionExerciseId`).

Furthermore, the collection exercise is an attribute of the casegroup and not the case. Therefore the request should probably be against the parent `/casegroup` resource.

There is already a `GET` method defined against the `casegroup` resource to retrieve a count of casegroups by `collectionExerciseId` so it would be more appropriate to have the `DELETE` method act on that.

However, it may be even more appropriate again to have the `DELETE` act on a new route called `/casegroups/collectionExercises/{collectionExerciseId}`